src/Aspire.Hosting/ExecutableResourceExtensions.cs

```diff
@@ -21,4 +21,53 @@ public static IEnumerable<ExecutableResource> GetExecutableResources(this Distri
 
         return model.Resources.OfType<ExecutableResource>();
     }
+
+    /// <summary>
+    /// Sets the command for the executable resource.
+    /// </summary>
+    /// <typeparam name="T">Type of executable resource.</typeparam>
+    /// <param name="builder">Builder for the executable resource.</param>
+    /// <param name="command">Command.</param>
+    /// <returns>The <see cref="IResourceBuilder{T}"/>.</returns>
+    public static IResourceBuilder<T> WithCommand<T>(this IResourceBuilder<T> builder, string command) where T : ExecutableResource
+    {
+        ArgumentException.ThrowIfNullOrEmpty(command);
+
+        var executableAnnotation = builder.Resource.Annotations.OfType<ExecutableAnnotation>().LastOrDefault();
+        if (executableAnnotation is { })
+        {
+            executableAnnotation.Command = command;
+        }
+        else
+        {
+            executableAnnotation = new ExecutableAnnotation
+            {
+                Command = command,
+                WorkingDirectory = string.Empty
+            };
+            builder.Resource.Annotations.Add(executableAnnotation);            
+        }
+
+        return builder;
+    }
+
+    /// <summary>
+    /// Sets the working directory for the executable resource.
+    /// </summary>
+    /// <typeparam name="T">Type of executable resource.</typeparam>
+    /// <param name="builder">Builder for the executable resource.</param>
+    /// <param name="workingDirectory">Working directory.</param>
+    /// <returns>The <see cref="IResourceBuilder{T}"/>.</returns>
+    public static IResourceBuilder<T> WithWorkingDirectory<T>(this IResourceBuilder<T> builder, string workingDirectory) where T : ExecutableResource
+    {
+        ArgumentNullException.ThrowIfNull(workingDirectory);
+
+        if (builder.Resource.Annotations.OfType<ExecutableAnnotation>().LastOrDefault() is { } executableAnnotation)
+        {
+            executableAnnotation.WorkingDirectory = workingDirectory;
```

Comment:
This needs to handle resolving absolute paths like this https://github.com/dotnet/aspire/blob/93577a15a46bb37c28871e27d4645389b31da062/src/Aspire.Hosting/ExecutableResourceBuilderExtensions.cs#L55

Reply:
There is still a path that will escape normalisation - creating the `ExecutableResource` directory / creating a derived resource type.

https://github.com/dotnet/aspire/blob/93577a15a46bb37c28871e27d4645389b31da062/tests/Aspire.Hosting.Tests/ExecutableResourceTests.cs#L114-L118

That is the current behaviour, so this Pr makes things no worse than before.  Could add a defensive path normalisation in `DcpExecutor` to cover it, although that does still leave the slight pain that the annotation is usually, but not always, normalised.